userspace/libsinsp/container_engine/cri.cpp

```diff
@@ -297,14 +297,28 @@ bool cri::resolve(sinsp_threadinfo *tinfo, bool query_os_for_missing_info)
 
 		if(!m_async_source)
 		{
-			uint64_t max_wait_ms = 10000;
+			// The maximum time to complete all attempts can be evaluated with
+			// the following formula, assuming at each round 2 CRI calls are
+			// performed with a deadline of 1000ms and that we use an
+			// exponential backoff with 125ms as start time:
+			//
+			// max_wait_ms = 2000 * n + 125 * (2^n - 1)
+			//
+			// Note that this excludes the time for the last 2 CRI API calls
+			// that will be performed anyway, even if the TTL expires.
+			//
+			// With n=5 the result is 13875ms, we keep some margin as we are
+			// taking into account elapsed time.
+			uint64_t max_wait_ms = 20000;
```

Comment:
AFAIK the deadline for the CRI calls is hardcoded: https://github.com/falcosecurity/libs/blob/0acc3ce10c0e0cd469d21889f114d2b461729d0d/userspace/libsinsp/cri.cpp#L54

Reply:
`s_cri_timeout` can be set from the lib client, via `sinsp::set_cri_timeout`, why do you say it's hardcoded?

Reply:
simply because I missed that 😄 , in this case, I agree it would be better to adjust the value automatically. Let's do it in a follow-up PR (maybe in the scope of https://github.com/falcosecurity/libs/pull/786)